tests/unit/test_sagemaker_steps.py

```diff
@@ -489,7 +493,10 @@ def test_training_step_creation_with_model(pca_estimator):
             'ExecutionRoleArn': EXECUTION_ROLE,
             'ModelName.$': "$['TrainingJobName']",
             'PrimaryContainer': {
-                'Environment': {},
+                'Environment': {
+                    'JobName': 'job_name',
+                    'ModelName': 'model_name'
+                },
```

Comment:
question: typically, i'm not a huge fan of modifying existing tests as it can mask regressions. In this case it seems reasonable, but I think we should also have a test where environment is not established in the estimator to verify that it still renders to empty. 

you're currently testing the `if` part of the control flow when environment is set, but are also taking away the test of a scenario where it isn't.

wdyt?

Reply:
There are tests that use a model without set environment variables (for example: [test_model_step_creation()](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/tests/unit/test_sagemaker_steps.py#L804) is using [pca_model](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/tests/unit/test_sagemaker_steps.py#L149) which doesn't have set environment variables)

I do agree that adding a test that tests the specific change would be better - will add it in next commit